From beac2fcf67fe3368e3e15b763a8ce054672583d7 Mon Sep 17 00:00:00 2001
From: Zhao Ye <zhao.ye@intel.com>
Date: Tue, 26 Sep 2023 12:24:09 +0800
Subject: [PATCH] Set security policy for iw command for Wifi ap mode

Add security policy for iw command for Wifi ap mode

Tracked-On: OAM-112365
Signed-off-by: Zhao Ye <zhao.ye@intel.com>
---
 private/genfs_contexts | 1 +
 private/iw.te          | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/private/genfs_contexts b/private/genfs_contexts
index 3d5c357e1..cebe4f9a1 100644
--- a/private/genfs_contexts
+++ b/private/genfs_contexts
@@ -173,6 +173,7 @@ genfscon sysfs /kernel/dmabuf/buffers u:object_r:sysfs_dmabuf_stats:s0
 genfscon sysfs /module/dm_verity/parameters/prefetch_cluster u:object_r:sysfs_dm_verity:s0
 genfscon sysfs /module/lowmemorykiller u:object_r:sysfs_lowmemorykiller:s0
 genfscon sysfs /module/tcp_cubic/parameters u:object_r:sysfs_net:s0
+genfscon sysfs /devices/pci0000:00/0000:00:1c.0/0000:02:00.0/ieee80211/phy0 u:object_r:sysfs_net:s0
 genfscon sysfs /module/wlan/parameters/fwpath u:object_r:sysfs_wlan_fwpath:s0
 genfscon sysfs /devices/virtual/timed_output/vibrator/enable u:object_r:sysfs_vibrator:s0
 genfscon sysfs /devices/virtual/misc/uhid u:object_r:sysfs_uhid:s0
diff --git a/private/iw.te b/private/iw.te
index f098e2286..ae92efc85 100644
--- a/private/iw.te
+++ b/private/iw.te
@@ -4,3 +4,6 @@ type iw_exec, system_file_type, exec_type, file_type;
 init_daemon_domain(iw)
 
 allow iw self:netlink_generic_socket create_socket_perms_no_ioctl;
+allow iw self:global_capability_class_set { sys_admin net_admin };
+allow iw sysfs_net:file r_file_perms;
+allow iw sysfs_net:dir r_dir_perms;
-- 
2.25.1

